### PR TITLE
Port Scallion to Scala 3

### PR DESCRIPTION
**Pull request by brunnerant**
This is a draft PR attempting to port Scallion to Scala 3. Currently, I just updated the dependencies on Silex and Scalatest to their latest available versions. I tried to port the code to Scala 3, but was stuck due to the slight differences with existential types. I tried replacing existential types by `Any`, but got stuck later on.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 
 val commonSettings = Seq(
   version            := "0.5",
-  scalaVersion       := "2.12.8",
-  crossScalaVersions := Seq("2.12.8", "2.13.1"),
+  scalaVersion       := "3.0.1",
+  crossScalaVersions := Seq("2.12.13", "2.13.4", "3.0.1"),
```
Comment (redelmann) [inline build.sbt]: Does it work on the specified versions? If not, please remove 2.x from `crossScalaVersions`.

File: build.sbt [hunk 2]
```diff
@@ -29,8 +29,8 @@ lazy val scallion = project
     target in Compile in doc := baseDirectory.value / "docs",
 
     libraryDependencies ++= Seq(
-      "org.scalatest" %% "scalatest" % "3.0.8" % "test",
-      "ch.epfl.lara" %% "silex" % "0.5" % "test",
+      "org.scalatest" %% "scalatest" % "3.2.9" % "test",
+      "ch.epfl.lara" %% "silex" % "0.6" % "test",
```
Comment (redelmann) [inline build.sbt:33]: Silex was made available on Bintray previously. Is there a convenient way to distribute it now?

**Review by redelmann — COMMENTED** [review]
Looking good, just a few remarks.

File: build.sbt (hunk 1 above)
Comment (brunnerant) [reply build.sbt]: Oh you're right, I wasn't aware `TypeTest` was not available in Scala 2.

File: build.sbt (hunk 2 above)
Comment (brunnerant) [reply build.sbt:33]: For now, we have to work with `publishLocal`...

Comment (LPTK) [reply build.sbt:33]: @redelmann Do you think the process for publishing it is different with Scala 3? Can't the new version of Silex be published as before? It's quite important that students not have to do `publishLocal` before compiling their projects.

Comment (redelmann) [reply build.sbt:33]: Bintray simply shut down. Better than publish local, simply put silex and scallion jars in a lib folder?

Comment (LPTK) [reply build.sbt:33]: Oh I see! Yes, some jars in a lib folder should do fine. Would be nice to publish on Sonatype/Maven Central, but that's a pain to set up.

Comment (hellotommmy) [reply build.sbt:33]: The current Scallion version I downloaded here gives the silex not found error when compiling.
Downloaded Silex as well and then publishLocal solved the issue.

**Comment by LPTK** [comment]
By the way @brunnerant, I opened a PR on your branch to bump the version to 0.6, as suggested earlier by @redelmann: https://github.com/brunnerant/scallion/pull/2

File: build.sbt (hunk 2 above)
Comment (redelmann) [reply build.sbt:33]: Yes, we need to include silex for the _examples_ project to work. I would suggest we add a lib folder for that specific project with the silex jar.

**Pull request merged**